engine/src/main/java/io/zeebe/engine/state/appliers/ProcessEventTriggeredApplier.java

```diff
@@ -22,7 +26,13 @@ public ProcessEventTriggeredApplier(final MutableEventScopeInstanceState eventSc
 
   @Override
   public void applyState(final long key, final ProcessEventRecord value) {
+    var variables = value.getVariablesBuffer();
+    if (variables.equals(DocumentValue.EMPTY_DOCUMENT)) {
+      // avoid storing an empty document
+      variables = NO_VARIABLES;
+    }
```

Comment:
Currently, the value has always no variable (i.e. it is used for error events only). But by default, it returns an empty document as variables.

Replace the empty document to avoid that it creates temporary variables.